File: dart/structural/decorator/guru_example/bin/concrete_decorator_a.dart
import 'component.dart';
import 'decorator.dart';

class ConcreteDecoratorA extends Decorator {
  ConcreteDecoratorA(Component componentValue) : super(componentValue);

  @override
  String operation() {
    return 'ConcreteDecoratorA(${super.operation()})';
  }
}

File: typescript/structural/decorator/guru_example/concrete_decorator_a.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConcreteDecoratorA = void 0;
const decorator_1 = require("./decorator");
class ConcreteDecoratorA extends decorator_1.Decorator {
    operation() {
        return `ConcreteDecoratorA(${super.operation()})`;
    }
}
exports.ConcreteDecoratorA = ConcreteDecoratorA;
